docs/proposals/scheduling/20220701-schedule-gang.md

```diff
@@ -0,0 +1,156 @@
+---
+title: gang scheduling
+authors:
+- "@buptcozy"
+reviewers:
+- "@hormes"
+- "@allwmh"
+- "@honpey"
+- "@jasonliu747"
+- "@saintube"
+- "@stormgbs"
+- "@zwzhang0107"
+- "@eahydra"
+creation-date: 2022-07-01
+last-updated: 2022-07-01
+status: provisional
+
+---
+
+# gang schedule
+
+## Table of Contents
+
+<!--ts-->
+
+* [gang schedule](#gang-schedule)
+  * [Table of Contents](#table-of-contents)
+  * [Summary](#summary)
+  * [Motivation](#motivation)
+  * [Goals](#goals)
+  * [User Stories](#user-stories)
+    * [Story AI](#story-AI)
+  * [Implementation Details](#implementation-details)
+    * [pod annotation to describe gang](#pod-annotation-to-describe-gang)
+    * [gang crd](#gang crd)
+    * [system-design](#system-design)
+
+<!--te-->
+
+## Summary
+gang scheduling
+
+## Motivation
+in AI scenario, we may want worker launch at same time.
+
+### Goals
+1. propose some pod's annotations to announce gang properties
```

Comment:
1. What's the difference between this proposl and [coscheduling](https://github.com/kubernetes-sigs/scheduler-plugins/blob/master/pkg/coscheduling/README.md)
2. What's the difference between this proposal and [volcano](https://github.com/volcano-sh/volcano)
3. Why do we need a CRD here? What's the pros and cons?
4. Why do we use annotations here? What's the pros and cons?
5. Could you elaborate each field in your CRD? So ppl would get a better idea.
6. What's the responsibilities of each stage in scheduling framework. PreFilter\Filter\Allocate\Permit.
7. A sequence diagram could explain a lot. Better have one.
8. What kind of YAML shoud user submit when he needs this feature? Could you give us a simple example?
9. Is there any limitations/alternatives of this design?

Reply:
updated